pkg/kubelet/cm/devicemanager/plugin/v1beta1/handler.go

```diff
@@ -106,8 +106,14 @@ func (s *server) deregisterClient(name string) {
 func (s *server) runClient(name string, c Client) {
 	c.Run()
 
-	c = s.getClient(name)
-	if c == nil {
+	curClient := s.getClient(name)
+	if curClient == nil {
+		return
+	}
+
+	// A new device plugin client has taken over and became the current.
+	if curClient.ID() != c.ID() {
```

Comment:
For same device-plugin this ID is same, eg. k8s-device-plugin socket value is `/var/lib/kubelet/device-plugins/nvidia-gpu.sock`. how to deal with this issue?

Reply:
As far as I understood this PR covers another scenario: new plugin should use different socket.